### PR TITLE
Hardcode -cuda extension to Lassen host-config to reflect switch to CUDA-only builds

### PR DESCRIPTION
**Pull request by joshessman-llnl**

### CODE REVIEW

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Review by jamiebramwell — APPROVED** [review]
This looked painful...

**Comment by joshessman-llnl** [comment]
LGTM

File: .gitlab/build_lassen.yml [hunk 1]
```diff
@@ -37,5 +37,8 @@ clang_upstream_gfortran (PR build_on_lassen):
 clang_upstream_gfortran (Main build_with_deps_on_lassen):
   variables:
     COMPILER: "clang@upstream_gfortran"
-    SPEC: "@develop%${COMPILER}"
+    SPEC: "@develop+cuda%${COMPILER}"
+    EXTRA_SPEC: "cuda_arch=70"
```
Comment (white238) [inline .gitlab/build_lassen.yml:41]: ANOTHER_VARIABLE_BECAUSE_GITLAB_CANT_HANDLE_SPACES_I_LOVE_MY_JOB

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**